cmd/tailscaled/tailscaled.go

```diff
@@ -903,6 +903,7 @@ func defaultEncryptState() bool {
 		// (plan9/FreeBSD/etc).
 		return false
 	}
-	v, _ := policyclient.Get().GetBoolean(pkey.EncryptState, false)
+	tpmSupported := hostinfo.New().TPM.Present()
```

Comment:
Running hostinfo.New() in init is a pretty expensive operation. It does things like probe your cloud environment, etc.

I'd rather the flag be a tristate (true/false/auto) rather than having to calculate true during init.

Then you could add a feature/hooks.go hook for probing the TPM, avoiding hostinfo.New